parsons/databases/redshift/redshift.py

```diff
@@ -420,32 +424,66 @@ def copy(self, table_obj, table_name, if_exists='fail', max_errors=0, distkey=No
                 See :ref:`parsons-table` for output options.
         """
 
-        # To Do:
-        # Auto split big files to copy more quickly
-        # Consider a json for better stability
-
-        key = self.temp_s3_copy(table_obj)
-
-        cols = None
+        # Specify the columns for a copy statement.
         if specifycols:
-            cols = table_obj.columns
+            cols = tbl.columns
+        else:
+            cols = None
 
-        try:
+        key = None
 
-            self.copy_s3(table_name, self.s3_temp_bucket, key,
-                         if_exists=if_exists, max_errors=max_errors,
-                         distkey=distkey, sortkey=sortkey, padding=padding, ignoreheader=1,
-                         varchar_max=varchar_max, statupdate=statupdate, compupdate=compupdate,
-                         acceptanydate=acceptanydate, dateformat=dateformat, timeformat=timeformat,
-                         blanksasnull=blanksasnull, nullas=nullas, emptyasnull=emptyasnull,
-                         acceptinvchars=acceptinvchars, truncatecolumns=truncatecolumns,
-                         columntypes=columntypes, specifycols=cols,
-                         aws_access_key_id=aws_access_key_id,
-                         aws_secret_access_key=aws_secret_access_key, compression='gzip')
+        with self.connection() as connection:
 
-        finally:
+            try:
```

Comment:
Minor thing, but the `try:` here is only really needed starting at line 460 (after the `temp_s3_copy` call), since the `finally:` is just about cleaning up the S3 object.. which won't exist until after the `temp_s3_copy` call.